Source/WebCore/platform/mediastream/RealtimeMediaSource.cpp

```diff
@@ -681,6 +659,37 @@ double RealtimeMediaSource::fitnessDistance(const MediaConstraint& constraint)
         break;
     }
 
+    case MediaConstraintType::WhiteBalanceMode: {
+        ASSERT(constraint.isString());
+        if (!capabilities.supportsWhiteBalanceMode())
+            return 0;
+
+        auto supportedModes = capabilities.whiteBalanceModes().map([](auto& mode) {
+            return convertEnumerationToString(mode);
+        });
+        return downcast<StringConstraint>(constraint).fitnessDistance(supportedModes);
+        break;
+    }
+
+    case MediaConstraintType::Zoom: {
+        ASSERT(constraint.isDouble());
+        if (!capabilities.supportsZoom())
+            return 0;
+
+        auto range = capabilities.zoom();
+        return downcast<DoubleConstraint>(constraint).fitnessDistance(range.rangeMin().asDouble, range.rangeMax().asDouble);
+        break;
+    }
+
+    case MediaConstraintType::Torch: {
+        ASSERT(constraint.isBoolean());
+        if (!capabilities.supportsTorch())
+            return 0;
+
+        return downcast<BooleanConstraint>(constraint).fitnessDistance(capabilities.torch());
```

Comment:
Given we are uncertain about required constraints for extensions such as torch et al, maybe we should just ignore them for now with regards to fitness distance computation, and add a FIXME?

Reply:
Good idea, I will do that